educe/learning/edu_input_format.py

```diff
@@ -28,13 +31,16 @@ def _dump_edu_input_file(docs, f):
             edu_txt = edu.text().replace('\n', ' ')
             # subgroup: sentence identifier, backoff on EDU id
             sent_idx = edu2sent[i]
-            subgroup = ('{}_sent{}'.format(grouping, sent_idx)
-                        if sent_idx is not None
-                        else edu_gid)
+            if sent_idx is None:
+                subgroup = edu_gid
+            elif isinstance(sent_idx, six.string_types):
```

Comment:
Well I mean they are “natural” in the sense that there's a pre-existing identifier for them which it would be quite natural to want to want to use.  

The issue isn't feature extraction time; I can just enumerate or whatever the turns.  It's more to do with the resulting .edu_input format file, you are referring to objects using an identifier scheme which you now have to know about.  

For example, a trivial example of this, is the attelo graph output which displays the subgrouping for each block of sentences.  OK that's a bit of a trivial case, but overall it's much more natural for other people working with these files to be able to select these turn objects from their already existing identifier rather than having to go through some mapping scheme.  

One option I was using initially was just to plug in the turn number outright, say turn “389” (these turn numbers can start fairly high)… which I think works for the purposes of learning, decoding etc; it just leads to these EDU input files that refer to objects that don't actually exist even though actually there's an identifier out there that works perfectly fine for them.

There's some interesting thinking about the numpy masked array stuff, but I wonder if edu_input_format is really the best level to be thinking about these things at.  I would tend to  imagine the _format modules to be fairly focused on the file formats leaving the higher level representational issues a bit more to the individual tasks (eg. STAC vs RST-DT)

As for renaming edu2subgroup, well I struggled somewhat inconsistently with this in attelo.  Sure we think of this as being intrasentential and many times I just couldn't resist the temptation to just say “sentence”… but in STAC it's plausible that intrasent style techniques could be straightforwardly applied to turns instead of sentences… in which case subgrouping would be the natural generalised name to cover both cases.  On the other hand; it would be less misleading to really just call them sentence if that's what we actually mean (if for example we had some assumptions baked in that are tied to sentences).

Overall though, I don't think it's overly general to try and cover the two projects you're currently working on :-)